packages/admin-ui/src/search/search.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import { createComponent, useElement, tag } from '@vtex/admin-ui-react'
+import { IconMagnifyingGlass, IconXCircle } from '@vtex/phosphor-icons'
+
+import * as styles from './search.style'
+import { Center } from '../center'
+import { Button } from '../button'
+import { SearchFormState } from './hooks/useSearchState'
+import { Spinner } from '../components/Spinner'
+import { VisuallyHidden } from '../components/VisuallyHidden'
+import { useMessageFormatter } from '../i18n'
+import { messages } from './search.i18n'
+
+const iconByName = {
+  search: <IconMagnifyingGlass size="small" csx={styles.icon} />,
+  spinner: <Spinner size={16} csx={styles.icon} />,
+}
+
+export const Search = createComponent<'form', SearchOptions>((props) => {
+  const formatMessage = useMessageFormatter(messages)
+
+  const label = formatMessage('search')
+
+  const {
+    state,
+    role = 'search',
+    'aria-label': ariaLabel = label,
+    disabled = false,
+    ...formProps
+  } = props
+
+  const iconName = state.loading ? 'spinner' : 'search'
```

Comment:
Just a comment not a change request I'm not sure what's the best approach... I'm always afraid of using string values for consts as these. But maybe typescript makes it safe, as the object won't accept other strings as key

Reply:
Yes, in this case, typescript makes it safe. I think it's worth discussing it, I'll let it this way, but if you want you can bring a discussion so the entire team can give an opinion about it

Reply:
I've changed to return the component instead of the strings because it was necessary to add a different style when the search is disabled